pallets/gear/src/manager/task.rs

```diff
@@ -132,21 +132,41 @@ where
 
     fn send_dispatch(&mut self, stashed_message_id: MessageId) {
         // No validation required. If program doesn't exist, then NotExecuted appears.
+
         let dispatch = DispatchStashOf::<T>::take(stashed_message_id)
             .unwrap_or_else(|| unreachable!("Scheduler & Stash logic invalidated!"));
 
+        // Unlock gas node to avoid of gas tree corruption
+        Pallet::<T>::unlock_gas_tree_node_if_exists(dispatch.id());
```

Comment:
why can't this be included directly inside `charge_gas_for_stashing`?

Reply:
and why `if_exist`? After your PR node should always exist

Reply:
make sure that on sendings not to mailbox message got consumed to avoid storage usage abuse

Reply:
Done